src/Version.php

```diff
@@ -97,7 +97,12 @@ public static function parse(string $version): static
             throw new InvalidVersionException('Invalid semantic version string provided');
         }
 
-        $version = sprintf('%s.%s.%s', $matches['major'], $matches['minor'] ?? 0, $matches['patch'] ?? 0);
+        $version = sprintf(
+            '%s.%s.%s',
+            $matches['major'],
+            isset($matches['minor']) && $matches['minor'] !== '' ? $matches['minor'] : 0,
+            isset($matches['patch']) && $matches['patch'] !== '' ? $matches['patch'] : 0,
+        );
```

Comment:
Instead of modifying this line, let's add the [`PREG_UNMATCHED_AS_NULL` flag](https://www.php.net/manual/en/pcre.constants.php#constant.preg-unmatched-as-null) to the `preg_match()` call above.

```php
if (! preg_match($semverRegex, $version, $matches, PREG_UNMATCHED_AS_NULL)) { ... }
```

This will cause unmatched segments to return `null` which will allow the null coalescing operator (i.e. `??`) to work as expected here.